src/lib/nvme_pt_ioctl.c

```diff
@@ -257,6 +258,12 @@ static void cpy_opalv200_feat(struct sed_opal_level0_discovery *discv,
 		sizeof(struct opalv200_supported_feat));
 }
 
+static void cpy_cnl_feat(struct sed_opal_level0_discovery *discv, void *feat)
+{
+	memcpy(&discv->sed_cnl, (struct cnl_feat *)feat,
+			sizeof(struct cnl_feat));
```

Comment:
to prevent accidental overflow typically size parameter in memcpy corresponds to dest paremeter not the src parameter. e.g.
memcpy(&discv->sed_cnl, (struct cnl_feat *) feat, sizeof(discv->sed_cnl))

Reply:
Yes! will do it